crates/cli-support/src/js/mod.rs

```diff
@@ -460,6 +461,37 @@ impl<'a> Context<'a> {
 
         self.unexport_unused_internal_exports();
         closures::rewrite(self)?;
+
+        // Handle the `start` function, if one was specified. If we're in a
+        // --test mode (such as wasm-bindgen-test-runner) then we skip this
+        // entirely. Otherwise we want to first add a start function to the
+        // `start` section if one is specified.
+        //
+        // Afterwards, we need to perfom what's a bit of a hack. Right after we
+        // added the start function, we remove it again because no current
+        // strategy for bundlers and deployment works well enough with it. For
+        // `--no-modules` output we need to be sure to call the start function
+        // after our exports are wired up (or most imported functions won't
+        // work).
+        //
+        // For ESM outputs bundlers like webpack also don't work because
+        // currently they run the wasm initialization before the JS glue
+        // initialization, meaning that if the wasm start function calls
+        // imported functions the JS glue isn't ready to go just yet.
+        //
+        // To handle `--no-modules` we just unstart the start function and call
+        // it manually. To handle the ESM use case we switch the start function
+        // to calling an imported function which defers the start function via
+        // `Promise.resolve().then(...)` to execute on the next microtask queue.
```

Comment:
Nitpick: the microtask queue is the queue of (micro) tasks, *not* a tick of the (micro) queue or a turn of the event loop, so this reads a bit funny to me.

```suggestion
        // `Promise.resolve().then(...)` to execute on the next microtask tick.
```